[aten] remove stmt unreachable, variable never used warnings

jspark1105 opened the pull request:
Summary:
Remove warning
```
caffe2/aten/src/THC/generic/THCTensorMathBlas.cu(437): warning: statement is unreachable
caffe2/aten/src/THC/generic/THCTensorMathBlas.cu(271): warning: variable "transpose_m1" was set but never used
caffe2/aten/src/THC/generic/THCTensorMathBlas.cu(271): warning: variable "transpose_m2" was set but never used
```

Test Plan: CI

Differential Revision: D20181179

ngimel commented on aten/src/THC/generic/THCTensorMathBlas.cu:
```diff
@@ -433,6 +435,7 @@ static void THCTensor_(addmmImpl)(THCState *state, THCTensor *r_, THCTensor *t,
                    r__->stride((transpose_r == 'n' ? 1 : 0)));
 #endif
 
+#if !defined(THC_REAL_IS_BFLOAT16) || defined(__HIP_PLATFORM_HCC__)
```
Why did these conditions need to change?

jspark1105 replied:
If these conditions do not met, we do TORCH_CHECK(false, ...) in line 419 and the compiler complains the statements below are unreachable.

jspark1105 replied:
That's a great point but TORCH_CHECK(false) is basically killing the process. Not sure what's PyTorch convention on whether we want to avoid leaks even when terminating the process.

ngimel replied:
TORCH_CHECK is throwing a runtime error that can be caught and handled.

jspark1105 replied:
Oh I see. I thought it's like assert. Thanks for letting me know and let me take a look to avoid leaks.

jspark1105 committed:
[aten] remove stmt unreachable, variable never used warnings (#34017)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34017

Remove warning
```
caffe2/aten/src/THC/generic/THCTensorMathBlas.cu(437): warning: statement is unreachable
caffe2/aten/src/THC/generic/THCTensorMathBlas.cu(271): warning: variable "transpose_m1" was set but never used
caffe2/aten/src/THC/generic/THCTensorMathBlas.cu(271): warning: variable "transpose_m2" was set but never used
```

Test Plan: CI

Differential Revision: D20181179

fbshipit-source-id: 2edc78a6d3ce4b7811295001ee0168e4931ae497
